apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/Optional.kt

```diff
@@ -18,6 +19,14 @@ sealed class Optional<out V> {
   object Absent : Optional<Nothing>()
 
   companion object {
+    @JvmStatic
+    fun <T> absent(): Optional<T> = Absent
```

Comment:
Might as well! [java.util.Optional(https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) (the implementation I am most familiar with) defines `of` and `ofNullable`

Reply:
From the javadoc of `of`:

```
Returns an Optional with the specified present non-null value.
```

That's really not what we want because we want to allow `present(null)` so I'd stick with:

```
Optional.absent()
Optional.present(value: T)
Optional.presentIfNotNull(value:T)
```

Reply:
Does `Optional.present(null)` make sense given that the parameter is defined to be not null? I think throwing a NullPointerException makes sense. (that's what `java.util.Optional#of` does)

Reply:
> given that the parameter is defined to be not null? 

The parameter is definitely nullable. If we wanted to enforce non-null, we'd put a `Any` upper bound:

```kotlin
// enforce non-null value (which we don't want)
fun <T: Any> present(value: T): Optional<T> = Present(value)

// allow null (which we want)
fun <T> present(value: T): Optional<T> = Present(value)
```

Reply:
Ah, my misunderstanding of generics in Kotlin I guess. I thought `T?` would allow null, and `T` would not.

Reply:
I guess the rationale is that `T` in itself can be of any Kotlin type, including nullable ones. It allows to do stuff like this:

```
fun <T:Any> assertNotNull(value: T?): T
```

So `T` without the `Any` upper bound can be nullable but once you put the `Any` upper bound, you have to type `T?` to make it nullable